src/useSession.tsx

```diff
@@ -81,3 +94,30 @@ export function useSession() {
     };
   }, []);
 }
+
+export const buildSessionUrlWithQueryParams = (
+  baseUrl: string,
+  landingPageUrl: string,
+  referrerUrl: string,
+): string => {
+  const sessionUrl = new URL(baseUrl);
+  sessionUrl.searchParams.set('landing_page_url', landingPageUrl);
+  sessionUrl.searchParams.set('referrer_url', referrerUrl);
+  return sessionUrl.toString();
+};
+
+export const resetLandingPageUrlQueryParams = (urlString: string): string => {
+  const url = new URL(urlString);
+  const params = url.searchParams;
+  const keysToRemove = [];
+
+  for (const key of params.keys()) {
+    if (key.toLowerCase().startsWith('utm_')) {
+      keysToRemove.push(key);
+    }
+  }
+
+  keysToRemove.forEach((key) => params.delete(key));
```

Comment:
Is it possible to delete the keys in a single loop as you find them? I'm not super familiar with the URL delete method so maybe not?

Reply:
good eye. Yes, no reason for this to be separated out especially since we already know if the key can be deleted in that conditional.